pkg/gadgets/trace/bind/tracer/tracer.go

```diff
@@ -15,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build linux
+// +build linux
```

Comment:
Instead of only relying on the OS, what about using a tag:
```
//go:build withebpf
// +build withebpf
```

So that kubectl-gadget can link with the gadget packages without including the ebpf code.

We used to have `withebpf` in the past (there are still some traces if you check with `git grep withebpf`).

The same approach was used for the `make generate-documentation` and the `docs` tag.

Reply:
Ah, great that it's already there to use. I didn't want to change more than needed in this PR, but we definitely have to do that before refactoring kubectl-gadget. Would you combine OS with the withebpf tag or leave that decision to the build system/Makefile?

Reply:
> Would you combine OS with the withebpf tag or leave that decision to the build system/Makefile?

I would leave it to the Makefile.

I would like the unit tests to continue to work on macosx when possible. Examples:
- ./pkg/gadgets/trace/dns/tracer/tracer_test.go --> the current test uses a function from tracer.go and that test works fine on macosx, so tracer.go should be compiled on macosx
- ./pkg/gadgets/trace/open/tracer/tracer_test.go --> the current tests require Linux so tracer_test.go has `//go:build linux`